### PR TITLE
skip compilation of rmw_cyclonedds when cyclone dds is not found

### PR DESCRIPTION
**Pull request by Karsten1987**
Title should be self explanatory. Basically don't compile rmw_cyclonedds_cpp when cyclonedds is not found on the system.

Signed-off-by: Karsten Knese <karsten@openrobotics.org>

### CODE REVIEW

File: rmw_cyclonedds_cpp/CMakeLists.txt [hunk 1]
```diff
@@ -32,7 +32,12 @@ find_package(ament_cmake_ros REQUIRED)
 find_package(rcutils REQUIRED)
 
 find_package(cyclonedds_cmake_module REQUIRED)
-find_package(CycloneDDS REQUIRED CONFIG)
+find_package(CycloneDDS QUIET CONFIG)
+if (NOT CycloneDDS_FOUND)
+  message(WARNING "Could not find cyclonedds - skipping '${PROJECT_NAME}'")
```
Comment (dirk-thomas) [inline rmw_cyclonedds_cpp/CMakeLists.txt]: Please use the proper case for the name and include the company name: `ADLINK CycloneDDS`.

File: rmw_cyclonedds_cpp/CMakeLists.txt [hunk 2]
```diff
@@ -32,7 +32,12 @@ find_package(ament_cmake_ros REQUIRED)
 find_package(rcutils REQUIRED)
 
 find_package(cyclonedds_cmake_module REQUIRED)
-find_package(CycloneDDS REQUIRED CONFIG)
+find_package(CycloneDDS QUIET CONFIG)
+if (NOT CycloneDDS_FOUND)
```
Comment (dirk-thomas) [inline rmw_cyclonedds_cpp/CMakeLists.txt]: CI has a linter warning for this line.

**Comment by Karsten1987** [comment]
My bad. Given that I don't build it locally on my machine, `colcon test` comes back with no errors given that no tests were running :)

File: rmw_cyclonedds_cpp/CMakeLists.txt [hunk 3]
```diff
@@ -32,7 +32,12 @@ find_package(ament_cmake_ros REQUIRED)
 find_package(rcutils REQUIRED)
 
 find_package(cyclonedds_cmake_module REQUIRED)
-find_package(CycloneDDS REQUIRED CONFIG)
+find_package(CycloneDDS QUIET CONFIG)
+if(NOT CycloneDDS_FOUND)
+  message(WARNING "Could not find ADLINK CycloneDDS - skipping '${PROJECT_NAME}'")
```
Comment (eboasson) [inline rmw_cyclonedds_cpp/CMakeLists.txt]: It isn't "ADLINK CycloneDDS": it is "Eclipse Cyclone DDS". The "Eclipse" bit really matters, but if it needs changing, then the space between "Cyclone" and "DDS" might as well be added, too.

(BTW: what is the appropriate way of dealing with such trivial changes? A comment like this, or just changing it in the editer? Just curious what in your experience is the best way to handle them.)

Comment (Karsten1987) [reply rmw_cyclonedds_cpp/CMakeLists.txt]: I just changed it to "Eclipse Cyclone DDS".
For changes like these, feel free to push directly on the branch to reflect your changes.

**Pull request merged**